yt_dlp/cookies.py

```diff
@@ -156,30 +156,22 @@ def _extract_firefox_cookies(profile, logger):
 
 
 def _firefox_browser_dir():
-    if sys.platform in ('linux', 'linux2'):
-        return os.path.expanduser('~/.mozilla/firefox')
-    elif sys.platform == 'win32':
-        return os.path.expandvars(R'%APPDATA%\Mozilla\Firefox\Profiles')
+    if sys.platform == 'win32':
+        profpath = os.path.expandvars(R'%APPDATA%\Mozilla\Firefox\Profiles')
     elif sys.platform == 'darwin':
-        return os.path.expanduser('~/Library/Application Support/Firefox')
+        profpath = os.path.expanduser('~/Library/Application Support/Firefox')
     else:
-        raise ValueError(f'unsupported platform: {sys.platform}')
+        profpath = os.path.expanduser('~/.mozilla/firefox')
+
+    if(os.path.exists(profpath)):
+        return profpath
+    else:
+        raise ValueError(f'cannot find profiles dir (unsupported platform: {sys.platform}?)')
```

Comment:
1. `_extract_chrome_cookies` etc checks whether the folder is found. It doesn't need to be checked here`
2. Windows can be `cygwin` as well

```suggestion
    if sys.platform in ('win32', 'cygwin'):
        return os.path.expandvars(R'%APPDATA%\Mozilla\Firefox\Profiles')
    elif sys.platform == 'darwin':
        return os.path.expanduser('~/Library/Application Support/Firefox')
    return os.path.expanduser('~/.mozilla/firefox')
```

Reply:
OK, Im always thinking "fail early", but let's not open that can of worms here.